### ISSUE TITLE
Bun 1.1.27 zlib regression

### ISSUE THREAD

**Issue by saadndm** [oven-sh/bun#14491]
### What version of Bun is running?

1.1.30+7996d06b8

### What platform is your computer?

Linux 6.11.0-63.fc41.x86_64 x86_64 unknown and Linux 6.6.31+rpt-rpi-2712 aarch64 unknown

### What steps can reproduce the bug?

With a bot created using this libary: https://github.com/PrismarineJS/mineflayer, reproducible every time, Error previously on `v1.1.29`:
```
huge pile of hexadecimal

error: incorrect header check
 code: "ERR_OPERATION_FAILED"


problem inflating chunk
uncompressed length 13942
compressed length 113
hex f66cfc8dc2fb1f5e3c77aed6f2a98ec6b8eb50fe4fe5a307e3cf6e88e7cfc68f46ff21ffc2dfc8bfcdfe2df66fb07f95fd63f68fd8bfc0fe3efbbbec6fb3bf99e2fffd20fa72f5fa0d71e16cfc50fcc187175f92fed17be4dfe97496947fbbdd5e50fead56aba9fc1b8d465df957abd58a
error: incorrect header check
 code: "ERR_OPERATION_FAILED"
```

and

Still getting this on 1.1.30
```ts
Error: Missing characters in string, found size is 117 expected size was 4219
problem inflating chunk
uncompressed length 235687
compressed length 24
hex a7b10effffff0f00000898fffffe3ec30001140018a6b10e
45 | export default "node:zlib";
                         ^
error: incorrect header check
 errno: -3
  code: "DataError"

      at zlibOnError (node:zlib:45:21)

^C
``````

### Additional information

Works fine when downgrading to bun `v1.1.26`:
`curl https://bun.sh/install | bash -s 'bun-v1.1.26'`

Every version after `v1.1.26` has this error

**Comment by not-nullptr** [comment]
reproducing with the latest mineflayer. anything on this?

**Comment by Jarred-Sumner** [comment]
Can you provide a script for us to run that reproduces the issue (without needing credentials or a Minecraft instance)? This is hard for us to reproduce.

**Comment by nektro** [comment]
thank you! :)

**Comment by BridgeSenseDev** [comment]
Fixed in #15757
